File: nom-api/Nom.Orch/Models/MealPlan/MealPlanRuleCreateResponseModel.cs
// File: Nom.Orch/Models/MealPlan/MealPlanRuleCreateResponseModel.cs

namespace Nom.Orch.Models.MealPlan
{
    public class MealPlanRuleCreateResponseModel
    {
        public long Id { get; set; }
        public long HouseholdId { get; set; }
        public long DayOfWeekId { get; set; }
        public long MealTypeId { get; set; }
        public string QueryFilterString { get; set; } = string.Empty;
        public DateTime CreatedDate { get; set; }
    }
} 

File: nom-ui/src/app/meal-plan/models/meal-plan-rule-create-response.model.ts
// File: nom-ui/src/app/meal-plan/models/meal-plan-rule-create-response.class.ts

import { IMealPlanRuleCreateResponseModel } from './meal-plan-rule-create-response.model.interface';

export class MealPlanRuleCreateResponseModel implements IMealPlanRuleCreateResponseModel {
    id = 0;
    name = '';
    description?: string;
    dayOfWeekId = 0;
    mealTypeId = 0;
    queryFilterString?: string;
    householdId = 0;
    isActive = true;
    createdDate: Date = new Date();

    constructor(data?: Partial<IMealPlanRuleCreateResponseModel>) {
        if (data) {
            Object.assign(this, data);
        }
    }
} 